server/util/lru/lru.go

```diff
@@ -114,14 +114,37 @@ func (c *LRU) Add(key, value interface{}) bool {
 	}
 
 	// Add new item
-	c.addItem(pk, ck, value)
+	c.addItem(pk, ck, value, true)
 
 	for c.currentSize > c.maxSize {
 		c.removeOldest()
 	}
 	return true
 }
 
+// PushBack adds a value to the back of the cache. Returns true if the key was added.
+func (c *LRU) PushBack(key, value interface{}) bool {
+	pk, ck, ok := c.keyHash(key)
+	if !ok {
+		return false
+	}
+	// Check for existing item
+	if ent, ok := c.lookupItem(pk, ck); ok {
+		c.evictList.MoveToFront(ent)
+		ent.Value.(*Entry).value = value
+		return true
+	}
+
+	// Add new item
+	c.addItem(pk, ck, value, false)
+
+	for c.currentSize > c.maxSize {
+		c.removeOldest()
+		return false
```

Comment:
note that because of the way the LRU evicts the oldest values first, if you PushBack a value and the LRU is full, that value will immediately be removed by this code and we'll return false.